components/base/wheeled/wheeled_base.go

```diff
@@ -396,3 +403,78 @@ func CreateWheeledBase(
 
 	return base, nil
 }
+
+type kinematicWheeledBase struct {
+	*wheeledBase
+	slam  slam.Service
+	model referenceframe.Model
+}
+
+func (base *wheeledBase) WrapWithKinematics(slam slam.Service) (base.KinematicBase, error) {
+	var err error
+	kwb := &kinematicWheeledBase{
+		wheeledBase: base,
+		slam:        slam,
+	}
+	kwb.model, err = kwb.buildModel(base.cfg)
+	return kwb, err
+}
+
+func (kwb *kinematicWheeledBase) ModelFrame() referenceframe.Model {
+	return kwb.model
+}
+
+func (kwb *kinematicWheeledBase) CurrentInputs(ctx context.Context) ([]referenceframe.Input, error) {
+	// TODO(RSDK-2311): complete the implementation
+	return []referenceframe.Input{}, errors.New("not implemented yet")
+}
+
+func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
+	// TODO(RSDK-2311): complete the implementation
+	return errors.New("not implemented yet")
+}
+
+func (kwb *kinematicWheeledBase) buildModel(cfg config.Component) (referenceframe.Model, error) {
+	// TODO(rb): this is a hacky workaround for not having kinematics for bases yet
+	// we create a sphere that would encompass the config geometry's rotation a full 360 degrees
+	// determine offset to use
+	geoCfg := cfg.Frame.Geometry
+	r := geoCfg.TranslationOffset.Norm()
```

Comment:
Instead of this, wouldn't it make sense to put the sphere at this offset?

Reply:
This is how he is handling the footprint during turning operations

Reply:
Ahhh, I see, this makes sense